lxd/images.go

```diff
@@ -3607,6 +3607,9 @@ func imageRefresh(d *Daemon, r *http.Request) response.Response {
 	// Begin background operation
 	run := func(op *operations.Operation) error {
 		_, err := autoUpdateImage(d.ctx, d, op, imageId, imageInfo, projectName, true)
+		if err != nil {
+			d.State().Events.SendLifecycle(projectName, lifecycle.ImageRefreshed.Event(fingerprint, projectName, op.Requestor(), nil))
```

Comment:
I wonder if `autoUpdateImage` returns us something to figure out whether a refresh actually happened, if so, we should only issue the event if an update was actually found.

Reply:
I believe we also want to issue this event for the scheduled image refreshes.

Reply:
I just moved the lifecycle event into the result check in `autoUpdateImage`.